File: Samples/CslaLight/cs/LocalDataPortalUsingWCFService/Web/ICompanyService.cs
//INSTANT C# NOTE: Formerly VB.NET project-level imports:
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Xml.Linq;
using System.Diagnostics;
using System.Collections.Specialized;
using System.Configuration;
using System.Text;
using System.Text.RegularExpressions;
using System.Web;
using System.Web.Caching;
using System.Web.SessionState;
using System.Web.Security;
using System.Web.Profile;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;

using System.ServiceModel;
using System.Runtime.Serialization;

namespace Web
{
	[ServiceContract()]
	public interface ICompanyService
	{

	  [OperationContract()]
	  UserInfo GetUser(string userName, string password);

	  [OperationContract()]
	  CompanyInfo GetCompany(int companyId);

	  [OperationContract()]
	  void UpdateCompany(CompanyInfo company);

	  [OperationContract()]
	  int InsertCompany(CompanyInfo company);

	  [OperationContract()]
	  void DeleteCompany(int companyId);

	}


	[DataContract()]
	public class UserInfo
	{
	  private string _userName = string.Empty;
	  private string _role = string.Empty;
	  private bool _isAuthenticated = false;

	  private UserInfo()
	  {
	  }

	  public UserInfo(string userName, string role, bool isAuthenticated)
	  {
		_userName = userName;
		_role = role;
		_isAuthenticated = isAuthenticated;
	  }

	  [DataMember()]
	  public bool IsAuthenticated
	  {
		get
		{
		  return _isAuthenticated;
		}
		set
		{
		  _isAuthenticated = value;
		}
	  }


	  [DataMember()]
	  public string UserName
	  {
		get
		{
		  return _userName;
		}
		set
		{
		  _userName = value;
		}
	  }

	  [DataMember()]
	  public string Role
	  {
		get
		{
		  return _role;
		}
		set
		{
		  _role = value;
		}
	  }
	}

	[DataContract()]
	public class CompanyInfo
	{
	  private int _companyId;
	  private string _companyName = string.Empty;
	  private string _dateAdded = string.Empty;


	  private CompanyInfo()
	  {
	  }

	  public CompanyInfo(int companyId, string companyName, string dateAdded)
	  {
		_companyId = companyId;
		_companyName = companyName;
		_dateAdded = dateAdded;
	  }

	  [DataMember()]
	  public int CompanyId
	  {
		get
		{
		  return _companyId;
		}
		set
		{
		  _companyId = value;
		}
	  }

	  [DataMember()]
	  public string CompanyName
	  {
		get
		{
		  return _companyName;
		}
		set
		{
		  _companyName = value;
		}
	  }

	  [DataMember()]
	  public string DateAdded
	  {
		get
		{
		  return _dateAdded;
		}
		set
		{
		  _dateAdded = value;
		}
	  }

	  public object DateAddedValue
	  {
		get
		{
		  if (string.IsNullOrEmpty(_dateAdded))
		  {
			return DBNull.Value;
		  }
		  else
		  {
			return _dateAdded;
		  }
		}
	  }

	}


} //end of root namespace

File: Samples/CslaLight/vb/LocalDataPortalUsingWCFService/Web/ICompanyService.vb
Imports System.ServiceModel
Imports System.Runtime.Serialization

<ServiceContract()> _
Public Interface ICompanyService

  <OperationContract()> _
  Function GetUser(ByVal userName As String, ByVal password As String) As UserInfo

  <OperationContract()> _
  Function GetCompany(ByVal companyId As Integer) As CompanyInfo

  <OperationContract()> _
  Sub UpdateCompany(ByVal company As CompanyInfo)

  <OperationContract()> _
  Function InsertCompany(ByVal company As CompanyInfo) As Integer

  <OperationContract()> _
  Sub DeleteCompany(ByVal companyId As Integer)

End Interface


<DataContract()> _
Public Class UserInfo
  Private _userName As String = String.Empty
  Private _role As String = String.Empty
  Private _isAuthenticated As Boolean = False

  Private Sub New()
  End Sub

  Public Sub New(ByVal userName As String, ByVal role As String, ByVal isAuthenticated As Boolean)
    _userName = userName
    _role = role
    _isAuthenticated = isAuthenticated
  End Sub

  <DataMember()> _
  Public Property IsAuthenticated() As Boolean
    Get
      Return _isAuthenticated
    End Get
    Set(ByVal value As Boolean)
      _isAuthenticated = value
    End Set
  End Property


  <DataMember()> _
  Public Property UserName() As String
    Get
      Return _userName
    End Get
    Set(ByVal value As String)
      _userName = value
    End Set
  End Property

  <DataMember()> _
  Public Property Role() As String
    Get
      Return _role
    End Get
    Set(ByVal value As String)
      _role = value
    End Set
  End Property
End Class

<DataContract()> _
Public Class CompanyInfo
  Private _companyId As Integer
  Private _companyName As String = String.Empty
  Private _dateAdded As String = String.Empty


  Private Sub New()
  End Sub

  Public Sub New(ByVal companyId As Integer, ByVal companyName As String, ByVal dateAdded As String)
    _companyId = companyId
    _companyName = companyName
    _dateAdded = dateAdded
  End Sub

  <DataMember()> _
  Public Property CompanyId() As Integer
    Get
      Return _companyId
    End Get
    Set(ByVal value As Integer)
      _companyId = value
    End Set
  End Property

  <DataMember()> _
  Public Property CompanyName() As String
    Get
      Return _companyName
    End Get
    Set(ByVal value As String)
      _companyName = value
    End Set
  End Property

  <DataMember()> _
  Public Property DateAdded() As String
    Get
      Return _dateAdded
    End Get
    Set(ByVal value As String)
      _dateAdded = value
    End Set
  End Property

  Public ReadOnly Property DateAddedValue() As Object
    Get
      If String.IsNullOrEmpty(_dateAdded) Then
        Return DBNull.Value
      Else
        Return _dateAdded
      End If
    End Get
  End Property

End Class

